julearn/model_selection/available_searchers.py

```diff
@@ -0,0 +1,96 @@
+from copy import deepcopy
+from julearn.utils.logging import raise_error, logger, warn
+from sklearn.model_selection import GridSearchCV, RandomizedSearchCV
+
+
+_available_searchers = {
+    'grid': GridSearchCV,
+    'random': RandomizedSearchCV
+}
+_available_searchers_reset = deepcopy(_available_searchers)
+
+
+def list_searchers():
+    """ List all available seraching algorithms
+
+    Returns
+    -------
+    out : list(str)
+        A list of all available seracher names.
+    """
+    return list(_available_searchers)
+
+
+def get_searcher(name):
+    """Get a searcher by name.
+
+    Parameters
+    ----------
+    name : str
+        The searchers name.
+
+    Returns
+    -------
+    out : obj
+        scikit-learn compatible searcher.
+    """
+    if name not in _available_searchers:
+        raise_error(
+            f'The specified searcher ({name}) is not available. '
+            f'Valid options are: {list(_available_searchers.keys())}'
+        )
+    out = _available_searchers[name]
+    return out
+
+
+def register_searcher(name, searcher, overwriting=None):
```

Comment:
Parameter should be called `overwrite`.  (maybe it's also like that in all the other `register`, should be changed).

Reply:
I changed for searchers in this PR. 
I will do the same for scorers in here #111 .
For transformers and estimators we will need another PR.